### PR TITLE
Added Wolfram Alpha presence

### PR DESCRIPTION
**Pull request by Kozou4ever**
I created Wolfram Alpha presence :)
Here are the proof the presence is working :

-https://i.imgur.com/PBJxtZu.png
-https://i.imgur.com/fPMZuSx.png
-https://i.imgur.com/AJXzJLl.png

### CODE REVIEW

**Review by EncryptedDev — COMMENTED** [review]
Probably should just check the domain for convention, but seems fine otherwise I guess

File: websites/W/Wolfram Alpha/presence.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/complex analysis")):
+            case (path.includes("/mathematical-function")):
+            case (path.includes("/logic-and-set-theory")):
+            case (path.includes("/mathematical-definitions")):
+            case (path.includes("/continued-fractions")):
+            case (path.includes("/famous-maths-problems")):
+            case (path.includes("/statistics")):
+            case (path.includes("/common-core-maths")):
+            case (path.includes("/probability")):
+                presenceData.details = "Reading for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:45]: Why you include `...` when you set state with title...

File: websites/W/Wolfram Alpha/presence.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                presenceData.state = title + " examples";
+                presenceData.smallImageKey = "reading";
+                presenceData.smallImageText = "Mathematics";
+                break;
+            default:
+                presenceData.details = "Looking for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:51]: Why you include `...` when you set state...

File: websites/W/Wolfram Alpha/presence.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/materials")):
+            case (path.includes("/transportation")):
+            case (path.includes("/technological-world")):
+            case (path.includes("/life-sciences")):
+            case (path.includes("/space-and-astronomy")):
+            case (path.includes("/weather-and-meteorology")):
+            case (path.includes("/physical-geography")):
+            case (path.includes("/health-and-medicine")):
+            case (path.includes("/food-science")):
+                presenceData.details = "Reading for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:71]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                presenceData.state = title + " examples";
+                presenceData.smallImageKey = "reading";
+                presenceData.smallImageText = "Science & Technology";
+                break;
+            default:
+                presenceData.details = "Looking for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:77]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/food-and-nutrition")):
+            case (path.includes("/demographics-and-social-statistics")):
+            case (path.includes("/institutions-and-organizations")):
+            case (path.includes("/art-and-design")):
+            case (path.includes("/political-geography")):
+            case (path.includes("/points-of-interest")):
+            case (path.includes("/games-and-puzzles")):
+            case (path.includes("/economic-data")):
+            case (path.includes("/education")):
+                presenceData.details = "Reading for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:97]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                presenceData.state = title + " examples";
+                presenceData.smallImageKey = "reading";
+                presenceData.smallImageText = "Society & Culture";
+                break;
+            default:
+                presenceData.details = "Looking for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:103]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/personal-finance")):
+            case (path.includes("/surprises")):
+            case (path.includes("/household-math")):
+            case (path.includes("/household-science")):
+            case (path.includes("/entertainment")):
+            case (path.includes("/travel")):
+            case (path.includes("/dates-and-times")):
+            case (path.includes("/todays-world")):
+            case (path.includes("/hobbies")):
+                presenceData.details = "Reading for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:118]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+                presenceData.state = title + " examples";
+                presenceData.smallImageKey = "reading";
+                presenceData.smallImageText = "Everyday life";
+                break;
+            default:
+                presenceData.details = "Looking for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:124]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+                presenceData.state = "Everyday life examples";
+                presenceData.smallImageKey = "reading";
+        }
+    }
+    else if (path.includes("/api")) {
+        presenceData.details = "Reading for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:130]: Why you include `...` when you set state.

File: websites/W/Wolfram Alpha/presence.ts [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+            if (theme.startsWith("A")) {
+                determinant = "an ";
+            }
+            presenceData.details = "Playing " + determinant + theme + " Quiz:";
+            presenceData.state = document.querySelector(".topicTitle").textContent.trim() +
+             " quiz | " + document.querySelector(".titleBar.Beginner").textContent.trim() + 
+             " | Correct: " + document.querySelector(".purpleBox").textContent.trim();
+        }
+        else {
+            presenceData.details = "Looking for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:151]: Why you include `...` when you set state.

**Review by ririxidev — CHANGES_REQUESTED** [review]

**Comment by Kozou4ever** [comment]
I don't see the point to change that but okay I guess

**Comment by EncryptedDev** [comment]
Bruh, since when was that a thing? It looks fine to have `...`

**Comment by callumok2004** [comment]
It looks better with `...` in my opinion, there was no reason to force Kozou to change it, it is not against the guidelines to use `...`, and after all, your job is ONLY to enforce said guidelines, and nothing more. Just cause you don't like it, don't mean Kozou can't have it

File: websites/W/Wolfram Alpha/presence.ts [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/complex analysis")):
+            case (path.includes("/mathematical-function")):
+            case (path.includes("/logic-and-set-theory")):
+            case (path.includes("/mathematical-definitions")):
+            case (path.includes("/continued-fractions")):
+            case (path.includes("/famous-maths-problems")):
+            case (path.includes("/statistics")):
+            case (path.includes("/common-core-maths")):
+            case (path.includes("/probability")):
+                presenceData.details = "Reading for:";
```
Comment (callumok2004) [inline websites/W/Wolfram Alpha/presence.ts:45]: Reading `for`? Simple Grammar error, remove the `for`

File: websites/W/Wolfram Alpha/presence.ts [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/materials")):
+            case (path.includes("/transportation")):
+            case (path.includes("/technological-world")):
+            case (path.includes("/life-sciences")):
+            case (path.includes("/space-and-astronomy")):
+            case (path.includes("/weather-and-meteorology")):
+            case (path.includes("/physical-geography")):
+            case (path.includes("/health-and-medicine")):
+            case (path.includes("/food-science")):
+                presenceData.details = "Reading for:";
```
Comment (callumok2004) [inline websites/W/Wolfram Alpha/presence.ts:71]: Reading `for`? Simple Grammar error, remove the `for`

File: websites/W/Wolfram Alpha/presence.ts [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/food-and-nutrition")):
+            case (path.includes("/demographics-and-social-statistics")):
+            case (path.includes("/institutions-and-organizations")):
+            case (path.includes("/art-and-design")):
+            case (path.includes("/political-geography")):
+            case (path.includes("/points-of-interest")):
+            case (path.includes("/games-and-puzzles")):
+            case (path.includes("/economic-data")):
+            case (path.includes("/education")):
+                presenceData.details = "Reading for:";
```
Comment (callumok2004) [inline websites/W/Wolfram Alpha/presence.ts:97]: Reading `for`? Simple Grammar error, remove the `for`

File: websites/W/Wolfram Alpha/presence.ts [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+                presenceData.smallImageText = "Everyday life";
+                break;
+            default:
+                presenceData.details = "Looking for:";
+                presenceData.state = "Everyday life examples";
+                presenceData.smallImageKey = "reading";
+        }
+    }
+    else if (path.includes("/api")) {
+        presenceData.details = "Reading for:";
```
Comment (callumok2004) [inline websites/W/Wolfram Alpha/presence.ts:130]: Reading `for`? Simple Grammar error, remove the `for`

File: websites/W/Wolfram Alpha/presence.ts [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+            case (path.includes("/personal-finance")):
+            case (path.includes("/surprises")):
+            case (path.includes("/household-math")):
+            case (path.includes("/household-science")):
+            case (path.includes("/entertainment")):
+            case (path.includes("/travel")):
+            case (path.includes("/dates-and-times")):
+            case (path.includes("/todays-world")):
+            case (path.includes("/hobbies")):
+                presenceData.details = "Reading for:";
```
Comment (callumok2004) [inline websites/W/Wolfram Alpha/presence.ts:118]: Reading `for`? Simple Grammar error, remove the `for`

**Review by callumok2004 — CHANGES_REQUESTED** [review]

File: websites/W/Wolfram Alpha/presence.ts (hunk 11 above)
Comment (Bas950) [inline websites/W/Wolfram Alpha/presence.ts:45]:
```suggestion
                presenceData.details = "Reading article:";
```

Comment (Bas950) [reply websites/W/Wolfram Alpha/presence.ts:45]:
```suggestion
                presenceData.details = "Reading about:";
```

Comment (Bas950) [reply websites/W/Wolfram Alpha/presence.ts:45]: Idk, for is not the correct thing at least xD

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/W/Wolfram Alpha/presence.ts (hunk 11 above)
Comment (Kozou4ever) [reply websites/W/Wolfram Alpha/presence.ts:45]: I might change it to "about", it looks good :)

**Review by Bas950 — APPROVED** [review]
LGTM

File: websites/W/Wolfram Alpha/presence.ts [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+    path = document.location.pathname;
+
+presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+        largeImageKey: "logo",
+        startTimestamp: browsingStamp
+    };
+
+    if (path.includes("/input")) {
+        presenceData.details = "Searching for...";
```
Comment (ririxidev) [inline websites/W/Wolfram Alpha/presence.ts:14]: didn't notice this one. you know what to do.

**Review by Bas950 — DISMISSED** [review]

**Comment by ririxidev** [comment]
cmon man, i didn't mean that. You had to change it to: ``Searching for:`` or ``Searching:`` not to ``Searching...``

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]
LGTM

**Pull request merged**